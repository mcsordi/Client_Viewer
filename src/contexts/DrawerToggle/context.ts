import { createContext } from 'react';
import { TNavDrawer } from '../../shared/types/NavDrawer';


type TToggle = {
    show: Boolean
    toggleState: () => void
    drawerOptions: TNavDrawer[]
    handleDrawerOptions: (navDrawer: TNavDrawer[]) => void
}

export const toggle = createContext({} as TToggle)
