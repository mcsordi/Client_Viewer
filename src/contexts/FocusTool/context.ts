import { createContext, Dispatch, SetStateAction } from "react";
type TFocusTool = {
    focus: boolean
    setFocus: Dispatch<SetStateAction<boolean>>
}

export const focusTool = createContext({} as TFocusTool)
