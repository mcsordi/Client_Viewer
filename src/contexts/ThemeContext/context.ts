import { createContext } from "react";

interface IThemeProvider {
    light: boolean,
    changeTheme: () => void
}

export const themeContext = createContext({} as IThemeProvider)

