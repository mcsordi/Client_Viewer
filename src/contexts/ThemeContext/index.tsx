import React, { useCallback, useState } from 'react';
import { themeContext } from './context.ts';
import { Children } from '../../shared/types/Children.ts';

export const ThemeContext: React.FC<Children> = ({ children }) => {
  const [light, setLight] = useState<boolean>(true);
  const changeTheme = useCallback(() => {
    setLight((old) => !old);
  }, []);
  return (
    <themeContext.Provider value={{ light, changeTheme }}>
      {children}
    </themeContext.Provider>
  );
};
