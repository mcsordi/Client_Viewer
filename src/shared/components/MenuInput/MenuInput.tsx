import React, { useContext, useMemo } from 'react';
import { themeContext } from '../../../contexts/ThemeContext/context';
import { Enviroment } from '../../../environment/index';
import { useSearchParams } from 'react-router-dom';

export const MenuInput: React.FC<{ visible?: boolean }> = ({
  visible = false,
}) => {
  const { light } = useContext(themeContext);
  const [searchParams, setSearchParams] = useSearchParams();

  const search = useMemo(() => {
    return searchParams.get('busca') || '';
  }, [searchParams]);

  {
    return (
      visible && (
        <input
          placeholder={Enviroment.INPUT_DE_BUSCA}
          type="search"
          value={search}
          className={`${
            light
              ? `bg-slate-200 placeholder:text-gray-500 border-gray-500 focus:bg-slate-300 transition-all
               focus:placeholder:text-gray-700`
              : `bg-zinc-800 text-white placeholder:text-white placeholder:opacity-70`
          }
             outline-none text-lg border rounded-md p-2 font-medium sm:w-96`}
          onChange={(e) =>
            setSearchParams({ busca: e.target.value }, { replace: true })
          }
        />
      )
    );
  }
};
