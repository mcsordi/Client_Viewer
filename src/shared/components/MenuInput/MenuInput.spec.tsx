import { render, screen } from '@testing-library/react';
import { MenuInput } from './MenuInput';
import { themeContext } from '../../../contexts/ThemeContext/context';
import { MemoryRouter } from 'react-router-dom';
import { Enviroment } from '../../../environment';

describe('<MenuInput/>', () => {
  it('should render a MenuInput component', () => {
    render(
      <MemoryRouter initialEntries={['/pessoas?busca=cavalo']}>
        <MenuInput visible />
      </MemoryRouter>,
    );
    const inputPlaceholder = screen.getByPlaceholderText(
      Enviroment.INPUT_DE_BUSCA,
    );
    expect(inputPlaceholder).toBeInTheDocument();
  });
  it('should render a MenuInput component without children', () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/pessoas?busca=cavalo']}>
        <MenuInput />
      </MemoryRouter>,
    );
    expect(container.children).toHaveLength(0);
  });
  it('should render a MenuInput component in the dark theme', () => {
    const changeTheme = () => {};
    render(
      <MemoryRouter initialEntries={['/pessoas?busca=cavalo']}>
        <themeContext.Provider value={{ light: false, changeTheme }}>
          <MenuInput visible />
        </themeContext.Provider>
      </MemoryRouter>,
    );
    const placeholder = screen.getByPlaceholderText(Enviroment.INPUT_DE_BUSCA);
    expect(placeholder).toHaveClass('bg-zinc-800 placeholder:text-white');
  });
  it('should render a MenuInput component in the white theme', () => {
    const changeTheme = () => {};
    render(
      <MemoryRouter initialEntries={['/pessoas?busca=cavalo']}>
        <themeContext.Provider value={{ light: true, changeTheme }}>
          <MenuInput visible />
        </themeContext.Provider>
      </MemoryRouter>,
    );
    const placeholder = screen.getByPlaceholderText(Enviroment.INPUT_DE_BUSCA);
    expect(placeholder).toHaveClass('bg-slate-200 placeholder:text-gray-500');
  });
  it('should match a snapshot', () => {
    const changeTheme = () => {};
    render(
      <MemoryRouter initialEntries={['/pessoas?busca=cavalo']}>
        <themeContext.Provider value={{ light: true, changeTheme }}>
          <MenuInput visible />
        </themeContext.Provider>
      </MemoryRouter>,
    );
    const placeholder = screen.getByPlaceholderText(Enviroment.INPUT_DE_BUSCA);
    expect(placeholder).toMatchSnapshot();
  });
});
