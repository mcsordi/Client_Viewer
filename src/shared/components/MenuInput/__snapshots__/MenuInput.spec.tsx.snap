// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuInput/> should match a snapshot 1`] = `
<input
  class="outline-none text-lg border rounded-md p-2 font-medium sm:w-96"
  placeholder="Pesquisar..."
  type="search"
/>
`;
